base/strings/search.jl

```diff
@@ -1,12 +1,14 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+@noinline __str_throw_boundserror(a, i) = throw(BoundsError(a, i))
+
 nothing_sentinel(i) = i == 0 ? nothing : i
 
 function findnext(pred::Fix2{<:Union{typeof(isequal),typeof(==)},<:AbstractChar},
                   s::String, i::Integer)
     if i < 1 || i > sizeof(s)
         i == sizeof(s) + 1 && return nothing
-        throw(BoundsError(s, i))
+        __str_throw_boundserror(s, i)
```

Comment:
I hope the compiler gurus can eventually fix having to manually include this non-intuitive  and ugly hack that is regrettably present in a lot of base function.

Reply:
I think this is mostly related to https://github.com/JuliaLang/julia/blob/0a685d3f62c34a3fc21040d118f01645435a415d/base/compiler/optimize.jl#L344-L349.

Reply:
Huh? There is no try/catch here.

Reply:
Sorry, my bad. I got confused.